devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java

```diff
@@ -125,14 +144,23 @@ public File getFastJar() {
     @TaskAction
     public void buildQuarkus() {
         final ApplicationModel appModel;
+        final List<Dependency> forcedDependencies = getForcedDependencies().getOrElse(Collections.emptyList()).stream()
+                .map(GACTV::fromString)
+                .map(a -> io.quarkus.maven.dependency.Dependency.of(a.getGroupId(), a.getArtifactId(), a.getVersion()))
+                .collect(Collectors.toList());
+
+        final Map<String, String> forcedProperties = getForcedProperties().getOrElse(Collections.emptyMap());
+
         try {
-            appModel = extension().getAppModelResolver().resolveModel(new GACTV(getProject().getGroup().toString(),
-                    getProject().getName(), getProject().getVersion().toString()));
+            appModel = extension().getAppModelResolver().resolveManagedModel(
+                    new GACTV(getProject().getGroup().toString(), getProject().getName(), getProject().getVersion().toString()),
+                    forcedDependencies, null, Collections.emptySet());
```

Comment:
The impl of this method in the Gradle plugin ignores the forced dependencies, which is why it doesn't work. I think I had a branch where I (tried to) implement it. We had some changes in main though and I'll need to rebase it and see if it still works.